presidio-analyzer/presidio_analyzer/predefined_recognizers/us_itin_recognizer.py

```diff
@@ -1,24 +1,45 @@
-from presidio_analyzer import Pattern
-from presidio_analyzer import PatternRecognizer
-
-# pylint: disable=line-too-long,abstract-method
-VERY_WEAK_REGEX = r'(\b(9\d{2})[- ]{1}((7[0-9]{1}|8[0-8]{1})|(9[0-2]{1})|(9[4-9]{1}))(\d{4})\b)|(\b(9\d{2})((7[0-9]{1}|8[0-8]{1})|(9[0-2]{1})|(9[4-9]{1}))[- ]{1}(\d{4})\b)'  # noqa: E501
-WEAK_REGEX = r'\b(9\d{2})((7[0-9]{1}|8[0-8]{1})|(9[0-2]{1})|(9[4-9]{1}))(\d{4})\b'  # noqa: E501
-MEDIUM_REGEX = r'\b(9\d{2})[- ]{1}((7[0-9]{1}|8[0-8]{1})|(9[0-2]{1})|(9[4-9]{1}))[- ]{1}(\d{4})\b'  # noqa: E501
-
-CONTEXT = [
-    "individual", "taxpayer", "itin", "tax", "payer", "taxid", "tin"
-]
+from presidio_analyzer import Pattern, PatternRecognizer
 
 
 class UsItinRecognizer(PatternRecognizer):
     """
     Recognizes US ITIN (Individual Taxpayer Identification Number) using regex
     """
 
-    def __init__(self):
-        patterns = [Pattern('Itin (very weak)', VERY_WEAK_REGEX, 0.05),
-                    Pattern('Itin (weak)', WEAK_REGEX, 0.3),
-                    Pattern('Itin (medium)', MEDIUM_REGEX, 0.5)]
-        super().__init__(supported_entity="US_ITIN", patterns=patterns,
-                         context=CONTEXT)
+    # pylint: disable=line-too-long,abstract-method
+    PATTERN_GROUPS = [
+        (
+            "Itin (very weak)",
+            r"(\b(9\d{2})[- ]{1}((7[0-9]{1}|8[0-8]{1})|(9[0-2]{1})|(9[4-9]{1}))(\d{4})\b)|(\b(9\d{2})((7[0-9]{1}|8[0-8]{1})|(9[0-2]{1})|(9[4-9]{1}))[- ]{1}(\d{4})\b)",  # noqa: E501
+            0.05,
+        ),
+        (
+            "Itin (weak)",
+            r"\b(9\d{2})((7[0-9]{1}|8[0-8]{1})|(9[0-2]{1})|(9[4-9]{1}))(\d{4})\b",  # noqa: E501
+            0.3,
+        ),
+        (
+            "Itin (medium)",
+            r"\b(9\d{2})[- ]{1}((7[0-9]{1}|8[0-8]{1})|(9[0-2]{1})|(9[4-9]{1}))[- ]{1}(\d{4})\b",  # noqa: E501
+            0.5,
+        ),
+    ]
+
+    CONTEXT = ["individual", "taxpayer", "itin", "tax", "payer", "taxid", "tin"]
```

Comment:
General remark about context: It is language dependent. Maybe it would be better to define it as a dictionary with language as key:
```
CONTEXT = {"en":["individual", "taxpayer", "itin", "tax", "payer", "taxid", "tin"],
                      "de":[]}
```

Then, during analysis, the requested language in the request would be used to fetch the relevant context. This allows us to use the same recognizer for multiple languages.

Reply:
I think you would need to redesign the whole system of recognizers to allow `supported_language` to be a set or list.  Right now, one needs to create a new instance of the recognizer for each language with the correct `supported_language`.  Also, I think maintaining a list of contexts in different languages would get unwieldy very quickly.  It is not difficult to make the change you are suggesting once, but I think you start to go down a path that will make maintenance a lot more difficult (i.e. adding a new recognizer would require adding context for every language and also making sure every recognizer had context for every supported language).

Reply:
In your opinion, what is the alternative? context terms are always language dependent. Either we have a recognizer per language (e.g. EnCreditCardRecognizer) or we allow a recognizer to accept context in multiple languages. Is there any additional alternative?

Reply:
So I would create a language specific recognizer using something like the following:

```python
il_domain_recognizer = DomainRecognizer(
    pattern_groups=[("IL Domain()", u"(www.)?[\u05D0-\u05EA]{2,63}.co.il", 0.5)],
    context=["תחום", "ip"],  # this is google translated
    supported_entity="IL_DOMAIN_NAME",
    supported_language="he"
)
```

Reply:
Overall it sounds good, however this would require the user to instantiate the recognizer with all parameters, as opposed to having some of them predefined within the class itself. WDYT?